src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -1272,20 +1121,21 @@ private static bool IsValidCallback(MethodInfo method, ParameterInfo[] parameter
                 return false;
             }
 
-            internal ClassDataContract? BaseContract
+            internal ClassDataContract? BaseClassContract
             {
-                get { return _baseContract; }
+                get => _baseContract;
                 set
                 {
                     _baseContract = value;
                     if (_baseContract != null && IsValueType)
-                        ThrowInvalidDataContractException(SR.Format(SR.ValueTypeCannotHaveBaseType, StableName!.Name, StableName.Namespace, _baseContract.StableName!.Name, _baseContract.StableName.Namespace));
+                        ThrowInvalidDataContractException(SR.Format(SR.ValueTypeCannotHaveBaseType, XmlName!.Name, XmlName.Namespace, _baseContract.XmlName!.Name, _baseContract.XmlName.Namespace));
                 }
             }
 
             internal List<DataMember>? Members
             {
-                get { return _members; }
+                get => _members;
+                set => _members = value;
```

Comment:
Should we internally do the same thing we expect to happen externally? E.g. keep this get only and the consumer that wants to set it can add to the existing instance and optionally clear it?

Reply:
I'm inclined not to. If we were designing this from scratch, maybe. But this has been how it has worked for ages. Doing something different slightly changes the semantics of how this works, which I'd rather not do in such a big PR. And to be fair, we don't actually intend for anyone to be using this other than our schema package... which doesn't do any writing to this collection.

Given that last point, maybe it's worth considering making the public exposure of this read-only?